### PR TITLE
(BKR-668) wrlinux5 puppet-agent install support added

### PR DESCRIPTION
**Pull request by kevpl**
Adds support for both puppet-agent install methods on wrlinux-5:
`install_puppet_agent_on` (from yum.puppetlabs.com)
`install_puppet_agent_dev_repo_on` (from builds.puppetlabs.net)

### CODE REVIEW

**Comment by kevpl** [comment]
ping @LuvCurves for test & review.

**Comment by kevpl** [comment]
:open_mouth: missed yard doc tests.  It was just a copied method signature comment that didn't get refactored correctly.  Updated, amended, & pushed.

**Comment by LuvCurves** [comment]
I have verified this works in the test environment like we expect.

**Comment by LuvCurves** [comment]
Found issue with call "on agent, puppet('--version')". Getting 2 prepends instead of 1
"source /etc/profile; sudo ip netns exec management source /etc/profile; sudo ip netns exec management puppet --version"

**Comment by kevpl** [comment]
@LuvCurves I've fixed the command prefix doubling.  Please test this out, and let me know if you're still seeing any issues there.

**Comment by LuvCurves** [comment]
scp is failing when it's done from test runner to cisco because it appears to be doing it with the 'root' account instead of 'beaker'

**Comment by kevpl** [comment]
@LuvCurves it's SCPing as beaker, but the files are owned by root at that time, that's why the permission failure.  I've put in a fix that will solve the issue for all SCPing to cisco hosts going forward.

For fun:

    Finished in 15.19 seconds (files took 2.51 seconds to load)
    1337 examples, 0 failures

**Comment by LuvCurves** [comment]
Nice work. You're correct, the file was owned by root and had permissions set to 600.  This last bit resolved my issue with the hiera acceptance test.

**Comment by kevpl** [comment]
@puppetlabs-jenkins retest this please

**Comment by kevpl** [comment]
@LuvCurves things look good to me from here.  I know your testing is ongoing, so let me know here what happens as you go through that, and please give a final :+1: when your testing passes so we can get this merged.  Thanks for your help testing this!

**Comment by kevpl** [comment]
killed the test myself, as I had a code update for that particular fix.  Will re-kick manually

**Comment by kevpl** [comment]
@puppetlabs-jenkins retest this please

**Comment by kevpl** [comment]
@puppetlabs-jenkins retest this please

**Comment by LuvCurves** [comment]
I've done some extensive testing with the latest changes and it's working great.

**Comment by kevpl** [comment]
We've found that the failure here is unrelated to these changes, as it's failing on master & in the modules pipelines in [this issue](https://tickets.puppetlabs.com/browse/BKR-695).

Since that's the case, this is ready for merge @puppetlabs/beaker!

File: lib/beaker/host.rb [hunk 1]
```diff
@@ -340,8 +340,8 @@ def exec command, options={}
     #
     #   do_scp_to('source/file.rb', 'target', { :ignore => 'file.rb' }
     #   -> will result in not files copyed to the host, all are ignored
-    def do_scp_to source, target, options
-      target = self.scp_path(target)
+    def do_scp_to source, target_path, options
+      target = self.scp_path( target_path )
```
Comment (waynr) [inline lib/beaker/host.rb]: yay for more semantically correct parameter names!

**Comment by waynr** [comment]
:+1: Looks good to me!

File: lib/beaker/host/unix/file.rb [hunk 2]
```diff
@@ -17,7 +17,13 @@ def system_temp_path
   #
   # @note This is really only needed in Windows at this point. Refer to
```
Comment (doug-rosser) [inline lib/beaker/host/unix/file.rb]: This comment seems odd, given the code below.

**Comment by doug-rosser** [comment]
Other than my quibble/question above, :+1:

**Comment by rick** [comment]
Only thing that stands out to me is additions of `/cisco/`, `/cisco-/`, and `/cisco-5/` -- is there a reason why one would be used over another, or are they all effectively equivalent?

**Comment by LuvCurves** [comment]
@rick There are actually 2 Cisco platforms we need to support cisco-wrlinux-5 (aka cisco-5) and cisco-wrlinux-7 (aka cisco-7).  I don't think we actually need /cisco/ but /cisco-/ would cover both platforms.

**Comment by kevpl** [comment]
@rick it should be that variant checks should be made against `cisco` itself, and platform checks should at least be against `cisco-`, unless the code really is specific for a version, hence `cisco-5`.

**Comment by rick** [comment]
:+1:

**Pull request merged**